app/src/main/kotlin/compiler/intermediate_form/ControlFlowGraphBuilder.kt

```diff
@@ -0,0 +1,42 @@
+package compiler.intermediate_form
+
+import compiler.common.reference_collections.ReferenceHashMap
+
+class ControlFlowGraphBuilder {
+    private val unconditionalLinks = ReferenceHashMap<IFTNode, IFTNode>()
+    private val conditionalTrueLinks = ReferenceHashMap<IFTNode, IFTNode>()
+    private val conditionalFalseLinks = ReferenceHashMap<IFTNode, IFTNode>()
```

Comment:
I think there is no possibility of single conditional link (neither true nor false), because in that case we wouldn't know how to continue the execution. In the example above, false link just leads to next instruction (just like the unconditional link at the end of `something`).

Reply:
In  `ControlFlow.kt`, line 95, it seems that such instructions you mentioned will create also conditional `False` link correctly, as @bebidek suggested.

Reply:
Okay, then let's verify that the conditional links have a corresponding pair during `build`